load-generator/src/main/java/com/amazon/opentelemetry/load/generator/emitter/StatsdMetricEmitter.java

```diff
@@ -43,24 +45,37 @@ public void emitDataLoad() throws Exception {
 
     @Override
     public void setupProvider() throws Exception {
+        log.info("Setting up Statsd metric emitter to generate load...");
         ipAddress = InetAddress.getByName(param.getEndpoint().split(":",2)[0]);
         portAddress = Integer.parseInt(param.getEndpoint().split(":",2)[1]);
         socket = new DatagramSocket();
+        log.info("Finished Setting up Statsd metric emitter.");
     }
 
     @Override
     public void nextDataPoint() {
         try {
             sendStatsd();
         } catch (IOException e) {
+            log.error("Error occurred while sending the packet!");
             e.printStackTrace();
         }
     }
 
     public void sendStatsd() throws IOException {
         byte buf[];
-        String payload = "statsdTestMetric1:1|g|#mykey:myvalue,mykey2:myvalue2";
-        buf = payload.getBytes();
+        StringBuilder payload;
+        payload = new StringBuilder();
+        String attributes = "#mykey:myvalue,datapoint_id:";
+        log.debug("Updating metrics...");
+        for (int i = 0; i < this.param.getMetricCount(); i++) {
+            for (int id = 0; id < this.param.getDatapointCount(); id++) {
+                payload.append("statsdTestMetric").append(i).append(":")
+                        .append(ThreadLocalRandom.current().nextInt(-100,100))
+                        .append("|g|").append(attributes).append(id).append("\n");
+            }
+        }
+        buf = payload.toString().getBytes();
         DatagramPacket packet = new DatagramPacket(buf, buf.length, ipAddress, portAddress);
```

Comment:
Just fyi, udp packets have size limitation of 64k. I think it is ok to let the exception bubble up, but maybe we can proactively check what is the size of `buf` before calling `send` and raise an `RuntimeException` asking the user to reduce the number of metrics/datapoints.

Reply:
Hi. Thanks updated.